coordinator/src/db/channels.rs

```diff
@@ -0,0 +1,153 @@
+use crate::schema;
+use crate::schema::channels;
+use crate::schema::sql_types::ChannelStateType;
+use anyhow::ensure;
+use anyhow::Result;
+use bitcoin::secp256k1::PublicKey;
+use diesel::query_builder::QueryId;
+use diesel::AsChangeset;
+use diesel::AsExpression;
+use diesel::BoolExpressionMethods;
+use diesel::ExpressionMethods;
+use diesel::FromSqlRow;
+use diesel::Insertable;
+use diesel::PgConnection;
+use diesel::QueryDsl;
+use diesel::QueryResult;
+use diesel::Queryable;
+use diesel::QueryableByName;
+use diesel::RunQueryDsl;
+use std::any::TypeId;
+use std::str::FromStr;
+use time::OffsetDateTime;
+
+#[derive(Debug, Clone, Copy, PartialEq, FromSqlRow, AsExpression)]
+#[diesel(sql_type = ChannelStateType)]
+pub enum ChannelState {
+    Pending,
+    Open,
+    Closed,
+    ForceClosedRemote,
+    ForceClosedLocal,
+}
+
+impl QueryId for ChannelStateType {
+    type QueryId = ChannelStateType;
+    const HAS_STATIC_QUERY_ID: bool = false;
+
+    fn query_id() -> Option<TypeId> {
+        None
+    }
+}
+
+#[derive(Insertable, QueryableByName, Queryable, Debug, Clone, PartialEq, AsChangeset)]
+#[diesel(table_name = channels)]
+pub struct Channel {
```

Comment:
🙃 I think it's cleaner to create a `Channel` in the models as well and not leak the database specific representation to the outside. This `Channel` struct should be private within `db::channels`. 
That way the database model can change independent of any business logic changes.